tests/unittests/CPUDeviceManagerTest.cpp

```diff
@@ -300,6 +300,83 @@ TEST(CPUDeviceManagerTest, MultiModule) {
   EXPECT_NE(ctx1, ctx2);
 }
 
+TEST(CPUDeviceManagerTest, ReuseModule) {
+  auto module = makeBasicModule("func1");
+
+  std::unique_ptr<Context> ctx1 = std::make_unique<Context>();
+  ctx1->allocate(module->getPlaceholders());
+
+  Function *F = module->createFunction("func2");
+  auto *input = module->getPlaceholderByName("input");
+  auto *C = F->createConv(*ctx1, "conv2a", input, 64, 4, 1, 0, 1);
+  ctx1->get(llvm::cast<Placeholder>(C->getFilter()))->getHandle().clear(0.3);
+  ctx1->get(llvm::cast<Placeholder>(C->getBias()))->getHandle().clear(0.4);
+  F->createSave("ret2", C);
+
+  std::vector<std::unique_ptr<CompiledFunction>> backing;
+  FunctionMapTy functions = compileFunctions(module.get(), backing);
+  EXPECT_EQ(functions.size(), 2);
+
+  // Split the function map into two parts.
+  FunctionMapTy functions2;
+  functions2.emplace("func2", std::move(functions["func2"]));
+  functions.erase("func2");
+  EXPECT_EQ(functions.size(), 1);
+  EXPECT_EQ(functions2.size(), 1);
+
+  CPUDeviceManager cpuCoreDevice;
+  cpuCoreDevice.init();
+
+  std::promise<const Module *> promise;
+  std::future<const Module *> future;
+  std::tie(promise, future) = getFutureHelper<const Module *>();
+  cpuCoreDevice.addNetwork(module.get(), std::move(functions),
+                           [&promise](const Module *module, ResultCode result) {
+                             callbackHelper(promise, module, result,
+                                            ResultCode::Ready);
+                           });
+  future.wait_for(2s);
+  EXPECT_EQ(future.get(), module.get());
+
+  std::tie(promise, future) = getFutureHelper<const Module *>();
+  cpuCoreDevice.addNetwork(module.get(), std::move(functions2),
+                           [&promise](const Module *module, ResultCode result) {
+                             callbackHelper(promise, module, result,
+                                            ResultCode::Ready);
+                           });
+  future.wait_for(2s);
```

Comment:
why do we need to wait for 2s and then future.get(). Can't it be just blocking future.get().

Reply:
It could, but in the event we have a bug in the test where the cb doesn't fire it'll hang. This way it just fails out in a couple seconds.